src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -27,25 +29,27 @@ import java.time.LocalTime
 import java.util.Optional
 
 @Service
+@Transactional // all public methods do updates, including getNotifications() !
 class NotificationService(
```

Comment:
now that service is transactional, these property getters would start transactions if they are not private